cardano-testnet/src/Testnet/Runtime.hs

```diff
@@ -167,9 +167,13 @@ startNode tp node ipv4 port _testnetMagic nodeCmd = GHC.withFrozenCallStack $ do
     -- We force the evaluation of initiateProcess so we can be sure that
     -- the process has started. This allows us to read stderr in order
     -- to fail early on errors generated from the cardano-node binary.
-    _ <- liftIO (IO.getPid hProcess)
+    pid <- liftIO (IO.getPid hProcess)
       >>= hoistMaybe (NodeExecutableError $ "startNode:" <+> pretty node <+> "'s process did not start.")
 
+    -- We then log the pid in the temp dir structure.
+    let nodePidFile = logDir </> node </> "node-" ++ show pid <.> "pid"
```

Comment:
It would be much better to store the pid inside the actual file e.g. `node.pid` instead in the file name

Reply:
It's easier to access. You just need to open `node.pid`. Otherwise you need to list files, parse the names and find the one with `node-*.pid` pattern if you're trying to do so programmatically.